ansible/roles/test/files/saitests/sai_qos_tests.py

```diff
@@ -58,21 +58,69 @@
 
 
 class ARPpopulate(sai_base_test.ThriftInterfaceDataPlane):
+    def setUp(self):
+        sai_base_test.ThriftInterfaceDataPlane.setUp(self)
+        time.sleep(5)
+        switch_init(self.client)
+
+        # Parse input parameters
+        self.router_mac = self.test_params['router_mac']
+        self.dst_port_id = int(self.test_params['dst_port_id'])
+        self.dst_port_ip = self.test_params['dst_port_ip']
+        self.dst_port_mac = self.dataplane.get_mac(0, self.dst_port_id)
+        self.src_port_id = int(self.test_params['src_port_id'])
+        self.src_port_ip = self.test_params['src_port_ip']
+        self.src_port_mac = self.dataplane.get_mac(0, self.src_port_id)
+        self.dst_port_2_id = int(self.test_params['dst_port_2_id'])
+        self.dst_port_2_ip = self.test_params['dst_port_2_ip']
+        self.dst_port_2_mac = self.dataplane.get_mac(0, self.dst_port_2_id)
+        self.dst_port_3_id = int(self.test_params['dst_port_3_id'])
+        self.dst_port_3_ip = self.test_params['dst_port_3_ip']
+        self.dst_port_3_mac = self.dataplane.get_mac(0, self.dst_port_3_id)
+
+    def tearDown(self):
+        sai_base_test.ThriftInterfaceDataPlane.tearDown(self)
+
     def runTest(self):
-        router_mac = self.test_params['router_mac']
-        # ARP Populate
-        index = 0
-        for port in ptf_ports():
-            arpreq_pkt = simple_arp_packet(
-                          eth_dst='ff:ff:ff:ff:ff:ff',
-                          eth_src=self.dataplane.get_mac(port[0],port[1]),
-                          arp_op=1,
-                          ip_snd='10.0.0.%d' % (index * 2 + 1),
-                          ip_tgt='10.0.0.%d' % (index * 2),
-                          hw_snd=self.dataplane.get_mac(port[0], port[1]),
-                          hw_tgt='00:00:00:00:00:00')
-            send_packet(self, port[1], arpreq_pkt)
-            index += 1
+         # ARP Populate
+        arpreq_pkt = simple_arp_packet(
+                      eth_dst='ff:ff:ff:ff:ff:ff',
+                      eth_src=self.src_port_mac,
+                      arp_op=1,
+                      ip_snd=self.src_port_ip,
+                      ip_tgt='192.168.0.1',
```

Comment:
I' m afraid to hardcoded the ip address as `192.168.0.1` will fail ptf topo. This ip address works only for t0.

Reply:
this is only needed for T0